// Generated by funapi_initiator
// This file is an example to illustrate how to use protobuf
// server-to-server messages on Funapi.
// You can edit this file to meet your taste.


// You should use a relative path to the system top-level directory.
// On Linux, it's /usr/include.
// To import /usr/include/funapi/network/fun_messages.proto,
// use relative path like this:
import "funapi/distribution/fun_rpc_message.proto";

message OnewayRpcMessage {
  optional string message = 1;
}

// Your specific message class.
message EchoRpcMessage {
  optional OnewayRpcMessage request = 1;
  optional OnewayRpcMessage reply = 2;
}

message OnewayMatchRpcMessage {
  optional string id = 1;
  optional string message = 2;
}

message MatchRpcMessage {
   optional OnewayMatchRpcMessage request = 1;
   optional OnewayMatchRpcMessage response = 2;
}

message MatchContextRpcMessage {
  optional string player_a = 1;
  optional string player_b = 2;
  optional OnewayMatchRpcMessage context = 3;
}

extend FunRpcMessage {
  optional OnewayRpcMessage oneway_rpc = 32;
  optional EchoRpcMessage echo_rpc = 33;
  optional OnewayMatchRpcMessage oneway_match_rpc = 34;
  optional MatchRpcMessage match_rpc = 35;
  optional MatchContextRpcMessage match_context_rpc = 36;
}
